### ISSUE TITLE
Allow custom roles for chip groups and chips

### ISSUE THREAD

**Issue by ienzam** [angular/components#15787]
#### Please describe the feature you would like to request.
MatChipList and MatChip do not support providing `role` attribute. Add support for that.

#### What is the use-case or motivation for this proposal?
I want to use MatChipList and MatChip as radiobox and radio respectively for better accessibility support. But can't override.

#### Is there anything else we should know?
Should be a trivial change. I can send a PR.

**Comment by ienzam** [comment]
Current hack I am using:
```
/** Hack to override role of MatChipList which is not supported natively. */
@Directive({selector: 'mat-chip-list[role]'})
export class MatChipListRoleOverride {
  @Input() @HostBinding('attr.role') role?: string|null;
}

/** Hack to override role of MatChip which is not supported natively. */
@Directive({selector: 'gmat-suggestive-chip[role]'})
export class MatChipRoleOverride {
  @Input() @HostBinding('attr.role') role?: string|null;
}
```

**Comment by jelbourn** [comment]
This will be addressed with the MDC-based chips implementation, which has `<mat-chip-set>` and `<mat-chip>` without roles.
